server/events/vcs/github_client.go

```diff
@@ -718,10 +719,27 @@ func StatusContextPassed(statusContext StatusContext, vcsstatusname string) bool
 }
 
 func ExpectedCheckPassed(expectedContext githubv4.String, checkRuns []CheckRun, statusContexts []StatusContext, vcsstatusname string) bool {
+	// If there's no WorkflowRun, we assume there's only one CheckRun with the given name.
+	// In this case, we evaluate and return the status of this CheckRun.
```

Comment:
This ought to be investigated further, but since the behaviour remains unchanged for this case, it shouldn't block this pull request. I might look into it at some point if I have the time.